src/main/java/org/generationcp/middleware/dao/AttributeDAO.java

```diff
@@ -251,23 +218,53 @@ public long countAttributesByGUID(final String germplasmUUID, final List<String>
 
 	private String buildQueryForAttributes(final List<String> attributeIds) {
 		String sql = "SELECT "
-			+ "    u.fcode AS attributeCode,"
-			+ "    u.fldno AS attributeDbId,"
-			+ "    u.fname AS attributeName,"
+			+ "    u.name AS attributeCode,"
+			+ "    u.cvterm_id AS attributeDbId,"
+			+ "    u.definition AS attributeName,"
 			+ "    a.adate AS determinedDate,"
 			+ "    a.aval AS value "
 			+ " FROM"
 			+ "    atributs a"
 			+ "        INNER JOIN"
-			+ "    udflds u ON a.atype = u.fldno "
+			+ "    cvterm u ON a.atype = u.cvterm_id "
 			+ "        INNER JOIN"
 			+ "    germplsm g ON a.gid = g.gid "
 			+ " WHERE"
-			+ "    g.germplsm_uuid = :germplasmUUID AND u.ftable = 'ATRIBUTS'";
+			+ "    g.germplsm_uuid = :germplasmUUID ";
 
 		if (attributeIds != null && !attributeIds.isEmpty()) {
-			sql = sql + " AND u.fldno IN ( :attributs )";
+			sql = sql + " AND u.cvterm_id IN ( :attributs )";
 		}
 		return sql;
 	}
+
+	public long countByVariable(final int variableId){
```

Comment:
@cuenyad I think maybe this function is not required if you use countByVariables and pass a List with only one id